### ISSUE TITLE
[EX-65] teach "task add" about context (automatically add relevant context criteria)

### ISSUE THREAD

**Issue by taskwarrior** [GothenburgBitFactory/taskwarrior#59]
_Scott Mcdermott on 2015-04-13T20:59:09Z says:_ 

 "context" feature is an automatic filter added to read-commands such as `list` depending on currently declared context.  This means one doesn't have to "remember" to always add `+work` for every `list`, just declare current context and this gets done automatically.

Same should be true for certain "write" commands (such as `add`).  Routine use of contexts in practice probably would mean that all one's tasks (including new ones) should get whatever attributes are being selected for in the current context.

Therefore, `task add` should be taught about context, and an option should exist to automatically append those attributes to new tasks.

**Comment by taskwarrior** [comment]
Migrated metadata:
```
Created: 2015-04-13T20:59:09Z
Modified: 2015-04-15T15:32:57Z
```

**Comment by taskwarrior** [comment]
_Scott Mcdermott on 2015-04-13T21:07:14Z says:_ 

 One idea is to have a configuration option that has a list of DOM element names which, if any such are present in the current context filter, will be automatically appended.  This lets ones which don't map from filter to writable attribute be ignored/unspecified.

you could store like this:

```
1.  current context
context=work
1.  definition of work context filter
context.work.filter=+work
1.  what DOM attributes from filter do we stash on 'add'
context.work.save=tags
```

**Comment by taskwarrior** [comment]
_Paul Beckingham on 2015-04-13T21:47:42Z says:_ 

 This is an ideal example for an `on-add` hook script.

**Comment by taskwarrior** [comment]
_David Patrick on 2015-04-14T17:25:43Z says:_ 

 I'm not sure that using context-elements for 'task add' is a good idea. 
If I'm at work, and I think of some random task _not_ related to work, I should be able to quickly add it without having to think about what context is active. (see: GTD directive #1)

That said, using taskwiki, and adding tasks under a ViewPort with user-defaults assigned, is very handy, but it is also very clear that these new tasks are part of this group.

The ability at have defaults assigned according to the current context is desirable, but it can't get in the way of a quick, unrelated task-add. Perhaps an alias command, like 'addcon', or some additional token ':@' or..

**Comment by taskwarrior** [comment]
_David Patrick on 2015-04-15T15:32:57Z says:_ 

 tasksh has a (mis-named) context function, that creates sticky filter elements, and that has the potential of serving this function well (default attributes on-add)

**Comment by pbeckingham** [comment]
No plans to implement.

**Comment by smemsh** [comment]
FYI to any passers-by, a similar capability was re-discussed in #1955 and is now implemented in 2.6.0 as read and write context variables via #2402.  There's a `context.foo.read` and `context.foo.write` now available in taskwarrior, the latter being used for eg `task add` and `task log`.  There was a follow-on discussion in freenode on 2015-06-10 where @tbabej mentioned the idea to have two separate config variables to suit this use case, and that appears to now be implemented.

**Comment by tbabej** [comment]
@smemsh Thanks for providing context! This improvement was indeed long overdue :slightly_smiling_face: I'm happy to ship this in 2.6.0!
